### PR TITLE
Add more configurability options

### PR DESCRIPTION
**Pull request by QuentinBisson**

### CODE REVIEW

File: helm/azure-ad-pod-identity-app/templates/nmi-daemonset.yaml [hunk 1]
```diff
@@ -26,26 +28,32 @@ spec:
...
       - name: nmi
         image: "{{ .Values.image.registry }}/{{ .Values.nmi.image.name }}:{{ .Values.nmi.image.tag }}"
         imagePullPolicy: {{ .Values.image.imagePullPolicy }}
         args:
           - "--host-ip=$(HOST_IP)"
           - "--node=$(NODE_NAME)"
+          - --http-probe-port={{ .Values.nmi.probePort | default 8080 }}
+          - --operation-mode={{ .Values.operationMode }}
+          {{- if eq .Values.operationMode "managed" }}
+          - --forceNamespaced
```
Comment (MarcelMue) [inline helm/azure-ad-pod-identity-app/templates/nmi-daemonset.yaml:54]: Why is this added when we run managed?

Comment (QuentinBisson) [reply helm/azure-ad-pod-identity-app/templates/nmi-daemonset.yaml:54]: It's what's used in capz and in the aad upstream chart:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/2e25580076af010974c0beb0fedd9d01155d36ce/config/default/aad-pod-identity-deployment.yaml#L110

Comment (MarcelMue) [reply helm/azure-ad-pod-identity-app/templates/nmi-daemonset.yaml:54]: Ohh okay - so "managed" means "like it is in CAPI"

**Review by MarcelMue — APPROVED** [review]

File: helm/azure-ad-pod-identity-app/templates/nmi-daemonset.yaml (hunk 1 above)
Comment (QuentinBisson) [reply helm/azure-ad-pod-identity-app/templates/nmi-daemonset.yaml]: Technically, standard means the mic component will assign then identities and managed that something else will. In our case, it should be capi yes
